code/Model/SiteTree.php

```diff
@@ -2755,6 +2764,21 @@ public function getControllerName() {
 	}
 
 	/**
+	 * Find the frontend controller name, which can be the current fully qualified class name
+	 * with Controller or _Controller added to it within the same namespace, or a customised
+	 * filename as per the $frontend_controller configuration option.
+	 *
+	 * @return string
+	 */
+	public function getFrontendControllerName()
+	{
+		if ($customController = $this->config()->frontend_controller) {
+			return (string) $customController;
+		}
+		return $this->getControllerName();
```

Comment:
Suggest `getControllerName()` is deprecated in favour of `getFrontendController()`. `getControllerName()` can return `get_class($this->getFrontEndController())` as deprecation shim, if needs be.